chain/indexer/src/streamer/streamer.rs

```diff
@@ -0,0 +1,230 @@
+    })
+}
+
+/// Fetches single chunk (as `near_primitives::views::ChunkView`) by provided `near_client::GetChunk` enum
+async fn fetch_single_chunk(
+    client: &Addr<near_client::ViewClientActor>,
+    get_chunk: near_client::GetChunk,
+) -> Result<views::ChunkView, FailedToFetchData> {
+    client.send(get_chunk).await?.map_err(|_| FailedToFetchData)
+}
+
+/// Fetch ExecutionOutcomeWithId for receipts and transactions from previous block
+/// Returns fetched Outcomes and Vec of failed to fetch outcome which should be retried to fetch
+/// in the next block
+async fn fetch_outcomes(
+    client: &Addr<near_client::ViewClientActor>,
+    mut outcomes_drain: impl Iterator<Item = types::TransactionOrReceiptId>,
+) -> (Vec<Outcome>, Vec<types::TransactionOrReceiptId>) {
+    let mut outcomes: Vec<Outcome> = vec![];
+    let mut outcomes_to_retry: Vec<types::TransactionOrReceiptId> = vec![];
+    while let Some(transaction_or_receipt_id) = outcomes_drain.next() {
+        match client
+            .send(near_client::GetExecutionOutcome { id: transaction_or_receipt_id.clone() })
+            .await
+            .map_err(|_| FailedToFetchData)
+        {
+            Ok(Ok(outcome)) => match &transaction_or_receipt_id {
+                types::TransactionOrReceiptId::Transaction { .. } => {
+                    outcomes.push(Outcome::Transaction(outcome.outcome_proof));
+                }
+                _ => {
+                    outcomes.push(Outcome::Receipt(outcome.outcome_proof));
+                }
+            },
+            _ => {
+                outcomes_to_retry.push(transaction_or_receipt_id);
+            }
+        }
+    }
+    (outcomes, outcomes_to_retry)
+}
+
+/// Fetches all the chunks by their hashes.
+/// Includes transactions and receipts in custom struct (to provide more info).
+/// Returns Chunks as a `Vec`
+async fn fetch_chunks(
+    client: &Addr<near_client::ViewClientActor>,
+    chunks: &[views::ChunkHeaderView],
+) -> Result<Vec<views::ChunkView>, FailedToFetchData> {
+    let chunks_hashes =
+        chunks.iter().map(|chunk| near_client::GetChunk::ChunkHash(chunk.chunk_hash.into()));
+    let mut chunks: futures::stream::FuturesUnordered<_> =
+        chunks_hashes.map(|get_chunk| fetch_single_chunk(&client, get_chunk)).collect();
+    let mut response = Vec::<views::ChunkView>::with_capacity(chunks.len());
+
+    while let Some(chunk) = chunks.next().await {
+        response.push(chunk?);
+    }
+
+    Ok(response)
+}
+
+/// Function that starts Streamer's busy loop. Every half a seconds it fetches the status
+/// compares to already fetched block height and in case it differs fetches new block of given height.
+///
+/// We have to pass `client: Addr<near_client::ClientActor>` and `view_client: Addr<near_client::ViewClientActor>`.
+pub async fn start(
+    view_client: Addr<near_client::ViewClientActor>,
+    client: Addr<near_client::ClientActor>,
+    mut queue: mpsc::Sender<BlockResponse>,
```

Comment:
Let's rename the `queue` into `blocks_sink`. (queues consist of two sides: source [pop/read] and sink [push/write])